internal/registration/internal/application/command_handler.go

```diff
@@ -0,0 +1,94 @@
+package application
+
+import (
+	"GoEdu/internal/registration/internal/domain"
+	"context"
+	"fmt"
+)
+
+type CommandHandler interface {
+	Handle(context.Context, Command) (CommandResult, error)
+}
+
+type RegisterNewUserCommandHandler struct {
+	hasher     PasswordHasher
+	verifier   UniqueEmailVerifier
+	repository UserRegistrationRepository
+}
+
+func (r *RegisterNewUserCommandHandler) Handle(context context.Context, command Command) (CommandResult, error) {
+	regNewUserCommand := command.(RegisterNewUserCommand)
+
+	h, err := r.hasher.Hash(regNewUserCommand.Password)
+	if err != nil {
+		return nil, err
+	}
+
+	user, err := domain.RegisterNewUser(
+		regNewUserCommand.FirstName,
+		regNewUserCommand.LastName,
+		regNewUserCommand.Email,
+		h,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if res, err := r.verifier.IsUnique(context, user.Email); err != nil || res != true {
+		if err != nil {
+			return nil, err
+		} else {
+			return nil, UserEmailMustBeUniqueError
+		}
+	}
+
+	err = r.repository.Add(context, *user)
+	if err != nil {
+		return nil, err
+	}
+
+	return user, nil
```

Comment:
Do not return a user for now. Instead I suggest you to create a new struct called RegisterNewUserCommandResult. For now it must be empty.